Adding extensions for expanding tilde in path and getting absolute path

macintosh-HD opened the pull request:
Base on the issue #7 and the recommendation of @tib I created two extensions to `String`:
- `expandingTildeInPath`
- `absolutePath`

I then used the two extensions in `Application+Static` to resolve the path from the environment and prevent errors.
By using `expandingTildeInPath` the `~` in the path is replaced with the users directory, preventing weird behaviour of the filesystem.
Additionally `absolutePath` makes sure that the path has a trailing slash. Otherwise new folders would be created one level beneath the wanted folder.

tib commented on Sources/App/Extensions/String+AbsolutePath.swift:
```diff
@@ -0,0 +1,18 @@
+import Foundation
+
+extension String {
+    public var absolutePath: String {
```
I believe that this function is not about building an absolute path, it just appends a trailing slash to a sting if it has no suffix. I don't really think that we need this extension to make things work for now.

macintosh-HD replied:
I added this function to prevent user error when reading the path from the environment. It happened to me that I accidentally forgot the trailing slash in the end of the path. The framework than just runs until it crashes because there are files missing. But it also creates a new folder below your working-directory.
I can understand if you don't like this specific solution, but I would suggest to prevent that user error or throw an actual error beforehand.

tib replied:
You are right about paths, maybe we should extend the Environment object with a static path() loader which could perform this job, also that could expand the tilde if necessary. What do you think? 

I'm going to merge this PR and we can open a new one for this issue if needed

tib commented on Sources/App/Extensions/String+ExpandingTilde.swift:
```diff
@@ -0,0 +1,19 @@
+    public var expandingTildeInPath: String {
+        var path = NSString(string: self).expandingTildeInPath
+        
+        if self.hasSuffix("/") {
```
Since we add the trailing slash within this function, the expandingTildeInPath method will return a correct value.

macintosh-HD replied:
This does not really add a slash. It only ensures that the output path has a trailing slash if the input path had a slash.
But if the original path had no slash, no slash gets added.

tib replied:
So as I mentioned this in the previous comment my idea is something like this:

Environment.path("BASE_PATH") -> this should return the proper path including the trailing slash & tilde resolution

```swift
extension Environment {
   static func path(_ key: String) -> String {
//      use .fetch
//      resolve ~ if necessary
//      append '/' if necessary
   }
}

If you agree with this approach (IMHO it's way better & safer than our original idea) could you please alter the PR one more time according to this so I can merge to master. Thanks again for your help. 👍

tib commented on Sources/App/Extensions/Application+Static.swift:
```diff
@@ -14,7 +14,7 @@ extension Application {
     // paths are always absolute, with a trailing slash
     struct Paths {
-        static let base: String = Environment.get("BASE_PATH")!
+        static let base: String = Environment.get("BASE_PATH")!.expandingTildeInPath.absolutePath
```
We don't need the .absolutePath here, see my other comments, also please merge with the latest master, since we have to use the `Environment.fetch` method from now on. Thank you for your help.

macintosh-HD commented:
I added the requested `path` extension for the `Environment` object.
For the extension I used the two `String` extensions I wrote earlier, but I made them only internally available. I also renamed the `absolutePath` variable to `withTrailingSlash`.
